modules/platform/nuxeo-platform-document-routing/nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/listener/DocumentRouteDeletedListener.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour AL KOTOB
+ */
+package org.nuxeo.ecm.platform.routing.core.listener;
+
+import static org.nuxeo.ecm.core.api.event.DocumentEventTypes.DOCUMENT_REMOVED;
+import static org.nuxeo.ecm.core.query.sql.NXQL.ECM_UUID;
+import static org.nuxeo.ecm.platform.routing.api.DocumentRoutingConstants.DOCUMENT_ROUTE_DOCUMENT_TYPE;
+import static org.nuxeo.ecm.platform.task.TaskConstants.TASK_PROCESS_ID_PROPERTY_NAME;
+
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.EventBundle;
+import org.nuxeo.ecm.core.event.PostCommitEventListener;
+import org.nuxeo.ecm.core.event.impl.DocumentEventContext;
+
+/**
+ * Listener that deletes orphan Tasks.
+ *
+ * @since 11.1
+ */
+public class DocumentRouteDeletedListener implements PostCommitEventListener {
+
+    protected static final String QUERY_GET_TASKS_RELATED_TO_DOCUMENT_ROUTE = "SELECT * FROM Document WHERE "
+            + TASK_PROCESS_ID_PROPERTY_NAME + " = '%s'";
+
+    @Override
+    public void handleEvent(EventBundle events) {
+        for (Event event : events) {
+            if (DOCUMENT_REMOVED.equals(event.getName())) {
+                DocumentEventContext context = (DocumentEventContext) event.getContext();
+                DocumentModel docModel = context.getSourceDocument();
+                if (DOCUMENT_ROUTE_DOCUMENT_TYPE.equals(docModel.getType())) {
+                    deleteOrphanTasks(context.getCoreSession(), docModel.getId());
+                }
+            }
+        }
+    }
+
+    /**
+     * Deletes all tasks which processId matches the given id.
+     *
+     * @param session the session used to query
+     * @param id the deleted DocumentRoute id
```

Comment:
I would remove the params and say in the main Javadoc "Deletes all tasks whose process id matches the given DocumentRoute id.".